src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -349,6 +360,10 @@ def _monitor_workers_and_process_tasks_in_thread(
 
             self._processing_idx.pop(line_number)
 
+        # If the while loop exits due to batch run timeout, we should set is_timeout to True.
+        if self._batch_timeout_expired(batch_start_time):
```

Comment:
It seems that if `self._is_timeout` is set to `True `in the monitor thread, it will be re-evaluated here. Additionally, if a batch run fails due to a batch timeout, it appears that this code will be executed regardless of whether task_queue is empty. Perhaps we can write it like this:
`self._is_timeout = self._is_timeout or self._batch_timeout_expired(batch_start_time)`

Reply:
Updated.